pkg/kv/kvserver/loqrecovery/collect.go

```diff
@@ -62,3 +76,68 @@ func CollectReplicaInfo(
 	}
 	return loqrecoverypb.NodeReplicaInfo{Replicas: replicas}, nil
 }
+
+// checkUncommittedDescriptorsInRaftLog iterates over committed but not applied
+// part of raft log and searches for writes to range descriptor. If any value
+// referencing local range descriptor key is found then we have a potential
+// pending update.
+func checkUncommittedDescriptorsInRaftLog(
+	rangeID roachpb.RangeID,
+	rangeKey roachpb.RKey,
+	startIndex uint64,
+	endIndex uint64,
```

Comment:
Sort of a convention at least in the raft code, which you may want to cargo cult here, is to call these `lo, hi uint64` where `lo` is inclusive, `hi` is exclusive (and note that in a comment).